openpype/hosts/max/plugins/publish/extract_thumbnail.py

```diff
@@ -17,34 +14,32 @@ class ExtractThumbnail(publish.Extractor):
     families = ["review"]
 
     def process(self, instance):
-        # TODO: Create temp directory for thumbnail
-        # - this is to avoid "override" of source file
-        tmp_staging = tempfile.mkdtemp(prefix="pyblish_tmp_")
-        self.log.debug(
-            f"Create temp directory {tmp_staging} for thumbnail"
-        )
-        fps = int(instance.data["fps"])
-        frame = int(instance.data["frameStartHandle"])
-        instance.context.data["cleanupFullPaths"].append(tmp_staging)
-        filename = "{name}_thumbnail..png".format(**instance.data)
-        filepath = os.path.join(tmp_staging, filename)
-        filepath = filepath.replace("\\", "/")
-        thumbnail = self.get_filename(instance.name, frame)
+        ext = instance.data.get("imageFormat")
+        frame = int(instance.data["frameStart"])
+        staging_dir = self.staging_dir(instance)
+        filepath = os.path.join(
+            staging_dir, f"{instance.name}_thumbnail")
+        self.log.debug("Writing Thumbnail to '{}'".format(filepath))
 
-        self.log.debug(
-            "Writing Thumbnail to"
-            " '%s' to '%s'" % (filename, tmp_staging))
         review_camera = instance.data["review_camera"]
-        with viewport_camera(review_camera):
-            preview_arg = self.set_preview_arg(
-                instance, filepath, fps, frame)
-            rt.execute(preview_arg)
+        viewport_options = instance.data.get("viewport_options", {})
+        files = render_preview_animation(
+            filepath,
+            ext,
+            review_camera,
+            start_frame=frame,
+            end_frame=frame,
+            width=instance.data["review_width"],
+            height=instance.data["review_height"],
+            viewport_options=viewport_options)
+
+        thumbnail = next(os.path.basename(path) for path in files)
 
         representation = {
             "name": "thumbnail",
             "ext": "png",
```

Comment:
Shouldn't this be `ext`?
```suggestion
            "ext": ext,
```